src/dune_rules/merlin/merlin.ml

```diff
@@ -466,18 +466,35 @@ module Unprocessed = struct
        } as t) sctx ~dir ~more_src_dirs ~expander =
     let open Action_builder.O in
     let+ config =
-      let* stdlib_dir, extra_obj_dirs =
+      let* stdlib_dir =
         Action_builder.of_memo
         @@
         match t.config.mode with
-        | `Ocaml -> Memo.return (Some stdlib_dir, [])
+        | `Ocaml -> Memo.return (Some stdlib_dir)
         | `Melange -> (
           let open Memo.O in
           let+ dirs = Melange_binary.where sctx ~loc:None ~dir in
           match dirs with
-          | [] -> (None, [])
-          | [ stdlib_dir ] -> (Some stdlib_dir, [])
-          | stdlib_dir :: extra_obj_dirs -> (Some stdlib_dir, extra_obj_dirs))
+          | [] -> None
+          | stdlib_dir :: _ -> Some stdlib_dir)
+      in
+      let* requires =
+        match t.config.mode with
+        | `Ocaml -> Action_builder.return requires
+        | `Melange ->
+          Action_builder.of_memo
+            (let open Memo.O in
+            let libs =
+              let scope = Expander.scope expander in
+              Scope.libs scope
+            in
+            Lib.DB.find libs (Lib_name.of_string "melange") >>= function
+            | Some lib ->
+              let+ libs =
+                Lib.closure [ lib ] ~linking:true |> Resolve.Memo.read_memo
```

Comment:
I think `~linking` should respect `implicit_transitive_deps` like it does elsewhere.

Also, instead of `read_memo`, should we just fail silently?

Reply:
I don't understand what you mean by failing silently here? We still need the `libs` to `Lib.Set.union` it to the other `requires`.

Reply:
do you mean I should just `Resolve.peek` and ignore if it's an `Error`?

Reply:
Pushed a version of the patch that uses `Resolve.Memo.peek` and takes `implicit_transitive_deps` into account

Reply:
Yeah, that's what I meant. I'm not even sure if it's the right thing to do, but it does seem a bit better to offer some support if some libraries fails to resolve.